### PR TITLE
Verify version and release for RPM PR test

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy [hunk 1]
```diff
@@ -17,3 +17,67 @@ def find_changed_packages(diff_range) {
...
+
+            for (int i = 0; i < packages.size(); i++) {
+                package_name = packages[i]
+                version_status = 'failed'
+                release_status = 'failed'
+
+                new_version = query_rpmspec("packages/${package_name}/*.spec", '%{VERSION}')
+                new_release = query_rpmspec("packages/${package_name}/*.spec", '%{RELEASE}')
+
+                sh "git checkout origin/${env.gitlabTargetBranch}"
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy]: Probably needs to use the GHPRB var here.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy [hunk 2]
```diff
@@ -17,3 +17,67 @@ def find_changed_packages(diff_range) {
...
+                      script: "rpmdev-vercmp ${old_release} ${new_release}",
+                      returnStatus: true
+                    )
+
+                    compare_new_to_one = sh(
+                      script: "rpmdev-vercmp 1 ${new_release}",
+                      returnStatus: true
+                    )
+
+                    if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy]: Edge case, but what if the version actually goes backwards? This can happen if a PR was merged in the mean time and an incorrect rebase.

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy]: Agreed, edge case not sure worth taking into account.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy [hunk 3]
```diff
@@ -53,6 +56,63 @@ pipeline {
...
+                              script: "rpmdev-vercmp ${old_release} ${new_release}",
+                              returnStatus: true
+                            )
+
+                            compare_new_to_one = sh(
+                              script: "rpmdev-vercmp 1 ${new_release}",
+                              returnStatus: true
+                            )
+
+                            if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy]: The `compare_new_to_one == VERCMP_OLDER` is to catch the release is set to 0.1?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy [hunk 4]
```diff
@@ -1,5 +1,8 @@
 def packages_to_build
 def packages = [:]
+def VERCMP_NEWER = 12
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy]: Should these be defined in `rpm_packaging.groovy` so every job that uses this can use them?

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy]: If this functionality were re-factored out I'd lean towards yes, but for now keeping them close to the code that uses them.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy (hunk 3 above)
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy]: I believe so.
